torch/fx/experimental/symbolic_shapes.py

```diff
@@ -847,10 +847,11 @@ def _constrain_range_for_size(a, min: Optional[int] = None, max: Optional[int] =
     if isinstance(a, (SymFloat, SymBool)):
         raise ValueError("Constraining SymFloat/SymBool is nyi")
 
-    assert isinstance(a, SymInt), "can only constrain range for SymInt"
-    assert isinstance(a.node.expr, sympy.Symbol), "constraining non-Symbols NYI"
-
-    a.node.shape_env._constrain_range_for_size(a.node.expr, min, max)
+    if (
+        isinstance(a, SymInt)
+        and isinstance(a.node.expr, sympy.Symbol)
+    ):  # follow _advise_is_size(), no-op if a is concrete value
+        a.node.shape_env._constrain_range_for_size(a.node.expr, min, max)
```

Comment:
While this probably is OK, it is a bit suspicious that you needed to no-op the asserts, because it suggests that maybe you might be using the wrong API downstream.